### ISSUE TITLE
Default COLOR_DARK_RED may soon no longer be "red"

### ISSUE THREAD

**Issue by xiota** [geany/geany#3016]
After #3013, the default color of `COLOR_DARK_RED` may no longer be "red".  Would there be any objection to changing the enum in [msgwindow.h](https://github.com/geany/geany/blob/6856edd6511e8aa9a225098ac2aab9171b10167f/src/msgwindow.h#L34-L40)?

* `COLOR_RED` → `COLOR_ERROR`
* `COLOR_DARK_RED` → `COLOR_ORANGE` → `COLOR_CONTEXT`
* `COLOR_BLUE` → `COLOR_MESSAGE`

The change would mostly be find/replace.  Since these values are exported, the old ones would need to be retained and marked as deprecated.  Plugins that would need to be updated: spellcheck, geniuspaste, project organizer, geanyctags, ¿geanypy?

From [`msgwindow.c`](https://github.com/geany/geany/blob/6856edd6511e8aa9a225098ac2aab9171b10167f/src/msgwindow.c#L296-L305):
```C
case COLOR_RED: return &color_error;
case COLOR_DARK_RED: return &color_context;
case COLOR_BLUE: return &color_message;
```

**Comment by elextr** [comment]
IIRC it was done this way to maintain backward compatibility when the CSS settings were added, they used to be hard coded colours.  Whilst I agree it would be nice to use a semantic name rather than a colour name its not important and it would have to be done in two steps.

1. make both the current and the new enumerators available in the enum but with the same value, and have the current enumerators deprecated for some period, then 
2. the current names can be removed.

**Comment by xiota** [comment]
> it was done this way to maintain backward compatibility when the CSS settings were added, they used to be hard coded colours.

I saw something about that mentioned in old issues/PRs.

> its not important

If not now, when?

It's a simple change that doesn't hurt anything.  I'll go ahead and open a PR after #3013 is merged.

* Internal Geany code would be updated in the same PR.
* Only a few plugins are affected that can be updated in a single PR.
* Even if the PR to update plugins is delayed, they would still build with the deprecated names for at least two Geany releases (1.40 or later). 
* Existing binaries should continue functioning.  (I'll test to confirm.)

**Comment by elextr** [comment]
> its not important

Like I said, its _nice_ but ultimately changing the name of the enumerator is not _required_, only humans will get confused by it, the compiler doesn't care what they are called.

AFAIK there is no process for removing something from the API, I certainly have never seen it happen.  The G-P collection plugins are not the only plugins, there are many others, known and unknown, out there.  Arbitary changes to the API has the potential to break them.

I suggested a process, with both names being available _with the same value_ and the old ones causing a deprecation message (assuming those are not turned off, so documentation needed as well).  

The normal evolution of things like this in software projects is to only do removals when a major version change happens.  It has been talked about doing a major version change (2.0) at the next release, but thats not enough time for random plugins out in the ecosystem to be updated, the deprecation needs to be propagated for some time.

**Comment by eht16** [comment]
I think this is OK when we, as already being said, keep the old color enums for backwards compability and issue a warning when they are being used.
As @xiota said, usage in Geany and G-P can be easily updated and for plugins out there in the wild, we set a deprecation period of a release or two, I don't think we need to tie it to major/minor version numbers, we are not a project like GTK or so.

**Comment by elextr** [comment]
> I don't think we need to tie it to major/minor version numbers, we are not a project like GTK or so.

True, we are much better, we never (deliberately) broke things :grin:

> the deprecation needs to be propagated for some time.
> we set a deprecation period of a release or two,

Seems a minimum, perhaps we should look at the other deprecated API, certainly some of the old plugin_init() family of functions are still used (eg commander) so those should stay.

**Comment by xiota** [comment]
I think the old plugin functions should stay a bit longer.  I've had unexplainable segfaults with the "new" plugin method that don't affect the "old" ones.  Just wrapping the functions with the old ones was sufficient to prevent the problem.

I don't doubt I was probably doing something wrong.  Why does the newer method seem more fragile?  Why is simply wrapping the functions sufficient to prevent the problem?

**Comment by elextr** [comment]
@xiota suggest you raise an issue with details of your problems.

**Comment by xiota** [comment]
Another approach would be to deprecate only COLOR_DARK_RED, replacing it with COLOR_ORANGE (and adding the semantic aliases).  This would affect far fewer plugins.  In geany-plugins, only geanypy would be affected, and it's already no longer usable.

**Comment by xiota** [comment]
> suggest you raise an issue with details of your problems.

I'll open an issue if I still have problems the next time I try (again) to migrate that plugin to the newer plugin method.
